plinth/utils.py

```diff
@@ -52,6 +52,11 @@ def non_admin_view(func):
     return func
 
 
-def is_user_admin(user):
+def is_user_admin(request, cached=False):
     """Return whether user is an administrator."""
-    return user.groups.filter(name='admin').exists()
+    if cached and request.session['cache_user_is_admin'] is not None:
```

Comment:
Also `return request.user.is_authenticated()` === `return False` in the above code.